be/src/storage/lake/general_tablet_writer.cpp

```diff
@@ -239,7 +254,17 @@ StatusOr<std::unique_ptr<SegmentWriter>> VerticalGeneralTabletWriter::create_seg
 
 Status VerticalGeneralTabletWriter::flush_columns(std::unique_ptr<SegmentWriter>* segment_writer) {
     uint64_t index_size = 0;
-    RETURN_IF_ERROR((*segment_writer)->finalize_columns(&index_size));
+    if (_async_write_segment) {
+        // Create a shared_ptr with a custom no-op deleter.
+        std::shared_ptr<SegmentWriter> shared_writer(segment_writer->get(), [](SegmentWriter*) {});
+        auto thread_pool_token = _segment_writer_finalize_token->get_thread_pool_token();
```

Comment:
if finalize_token wraps the ThreadPoolToken, why not add a `submit_func` interface, so the ThreadPoolToken is completely hides behind the finalizeTooken.